# Assignment 6

To measure the psychological state of Americans, a psychology researcher (in 1929) devised
three questions put to 3443 Americans from different walks of life. He asked them whether they
were happy. They could answer only **yes** or **no** to each question. By today’s standards the study
is quite simple, and he did no formal statistical analysis of his data. The frequency data (number
of **yes** and **no** responses within each employment category) from this study are in the file
*Americandata.csv*.

---

You need to use the data and answer the question, that is, **Is the type of profession related to the happiness of American people?**
Report the **χ2** statistic and the *statistical significance of test*, in addition to the contingency table.
Also, explain the results and what they mean. What does the value of **χ2** statistic mean? Is it
good or bad? Is the test statistically significant?

In [1]:
import pandas as pd
from scipy.stats import chi2_contingency

df = pd.read_csv('Americandata.csv', index_col='Happy')
chi2, p, dof, expected = chi2_contingency(df)

print("Contingency table")
print(df)
print()
print("Expected frequencies")
print(pd.DataFrame(expected, index=df.index, columns=df.columns).round(1))
print()
print(f"Chi-square = {chi2:.2f}, df = {dof}, p-value = {p:.3e}")

Contingency table
       Students  Laborers  Preachers  Physicians  Housewives  Teachers  \
Happy                                                                    
Yes         390       378         35         159          78       108   
No         1610       122        265          51         122        38   

       Lawyers  Musicians  
Happy                      
Yes         11         31  
No          64         19  

Expected frequencies
       Students  Laborers  Preachers  Physicians  Housewives  Teachers  \
Happy                                                                    
Yes       683.7     170.9      102.6        71.8        68.4      49.9   
No       1316.3     329.1      197.4       138.2       131.6      96.1   

       Lawyers  Musicians  
Happy                      
Yes       25.6       17.1  
No        49.4       32.9  

Chi-square = 936.14, df = 7, p-value = 7.523e-198


Put your analysis in the empty `markdown` cell below.

I ran a chi-square test to see if profession and happiness are related, and I got χ² = 936.14 with 7 degrees of freedom. That number is huge, which basically means the real counts are way off from what we'd expect if job and happiness had nothing to do with each other. The p-value (7.523e-198) is super tiny compared to 0.05, so the test is significant and profession and happiness are definitely connected.

Looking at the tables, laborers, physicians, and teachers said yes a lot more than expected, while students, preachers, and lawyers said yes a lot less. I don't think this means a job causes happiness though, it just shows that the two are linked.